Framework/PythonInterface/mantid/plots/plotfunctions.py

```diff
@@ -98,10 +99,24 @@ def _plot_impl(axes, workspace, args, kwargs):
         kwargs['linestyle'] = 'steps-post'
     else:
         x, y, _, _, indices, kwargs = _get_data_for_plot(axes, workspace, kwargs)
-        _setLabels1D(axes, workspace, indices)
+        try:
+            plot_as_distribution = (on_off_to_bool(config['graph1d.autodistribution']) and
+                                    not workspace.isDistribution())
```

Comment:
Do we need to consider the distribution state of the workspace now if that is all encapsulated in the `MatrixWorkspace`?